api/src/main/kotlin/com/google/devtools/ksp/utils.kt

```diff
@@ -99,7 +99,9 @@ fun KSDeclaration.isLocal(): Boolean {
  * Perform a validation on a given symbol to check if all interested types in symbols enclosed scope are valid, i.e. resolvable.
  * @param predicate: A lambda for filtering interested symbols for performance purpose. Default checks all.
  */
-fun KSNode.validate(predicate: (KSNode?, KSNode) -> Boolean = { _, _-> true } ): Boolean {
+fun KSNode.validate(predicate: (KSNode?, KSNode) -> Boolean = { parent, current ->
+    !(parent is KSFunctionDeclaration && current is KSDeclaration)
+} ): Boolean {
```

Comment:
Can we simply make `visitFunctionDeclaration` not calling `visitDeclarationContainer`?

Reply:
I want to leave the door open for those who still want to check for function body.